daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ScriptF.scala

```diff
@@ -281,7 +281,7 @@ object ScriptF {
 
       for {
         viewType <- Converter.toFuture(env.lookupInterfaceViewTy(interfaceId))
-        client <- Converter.toFuture(env.clients.getPartyParticipant(parties.head))
```

Comment:
There remains two other instances of `getPartyParticipant(parties.head)`...
in `queryContractId` and `queryViewContractId`

Are these also wrong?

Reply:
Oh, I didn't know `queryContractId` also used that. I'm pretty sure these are all wrong, but to test that we'd need to set up an environment where a daml script uses e.g. `queryContractId` with two parties on different participants, and I haven't done anything like that.

Reply:
to be clear, with `getPartiesParticipant(parties)` the command would just fail in such a case, whereas with `getPartyParticipant(parties.head)` it will continue, likely returning only the contracts visible to the subset of the parties on that participant.

https://github.com/digital-asset/daml/blob/a16fd640c21b086c0f7c239795700c95bf95d2d7/daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala#L90-L105

Reply:
We have 1 multi-participant test for script.
```
//daml-script/test:multi-participant-integration-test
```

Reply:
It seems wrong/odd that `query` and friends use `getPartiesParticipant`, given the requirement that all parties reside on the same participant.

Semantically (at least in `IdeLedgerClient`) the multiple `parties` are interpreted disjunctively. That is: only one of the parties needs be a stakeholder for the contract-instance to be deemed visible to the query: `parties.any(stakeholders.contains(_)`

Given this interpretation, I claim it would be more natural for a daml-script `query` with multiple parties to delegate to *all* participants of all the parties, and then merge the results.

Reply:
that's a very good point @nickchapman-da. The docs for the relevant functions in `daml-script` always use "party" in the singular, so the semantics these are meant to have are not clear to me. If we go by the existing `IdeLedgerClient` code I agree that the logical thing to do would be to merge the results

Reply:
> It seems wrong/odd that query and friends use getPartiesParticipant, given the requirement that all parties reside on the same participant.

Parties does **not** reside on the same participant, see the multi-participant tests. 

> at least in IdeLedgerClient

`IdeLedgerClient` emulates a single participant ledger. To emulate a multi-partipant ledger we would need to project transactions, however there is nothing in the main repo to do that.